src/main/java/org/opensearch/knn/training/FloatTrainingDataConsumer.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.training;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.knn.jni.JNIService;
+import org.opensearch.knn.index.memory.NativeMemoryAllocation;
+import org.opensearch.search.SearchHit;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Transfers float vectors from JVM to native memory.
+ */
+public class FloatTrainingDataConsumer extends TrainingDataConsumer {
+
+    /**
+     * Constructor
+     *
+     * @param trainingDataAllocation NativeMemoryAllocation that contains information about native memory allocation.
+     */
+    public FloatTrainingDataConsumer(NativeMemoryAllocation.TrainingDataAllocation trainingDataAllocation) {
+        super(trainingDataAllocation);
+    }
+
+    @Override
+    public void accept(List<?> floats) {
+        trainingDataAllocation.setMemoryAddress(
+            JNIService.transferVectors(
+                trainingDataAllocation.getMemoryAddress(),
+                floats.stream().map(v -> ArrayUtils.toPrimitive((Float[]) v)).toArray(float[][]::new)
+            )
+        );
+    }
+
+    @Override
+    public void processTrainingVectors(SearchResponse searchResponse, int vectorsToAdd, String fieldName) {
```

Comment:
Is there a unit test for this method?

Reply:
Yes, in [VectorReaderTests.java](https://github.com/opensearch-project/k-NN/blob/5685992f6b3266b34eac2c76576a59318461eea2/src/test/java/org/opensearch/knn/training/VectorReaderTests.java) by TestFloatTrainingDataConsumer.

Reply:
I think, we should name it as` FloatTrainingDataConsumerTests`

Reply:
There is a existing test class named `FloatTrainingDataConsumerTests`, but I think VectorReaderTests can cover the test of processTrainingVectors. Will have another PR for the refactor.